### PR TITLE
use kafka for publishing metrics

### PR DESCRIPTION
**Pull request by Dieterbe**
not tested yet. but compiles :)

also @woodsaj can you give me push access to this repo.

### CODE REVIEW

File: metric_publish/publish.go [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
-	cfg.UserAgent = fmt.Sprintf("tsdb-gw-server")
-	var err error
-	globalProducer, err = nsq.NewProducer(addr, cfg)
+	// We are looking for strong consistency semantics.
+	// Because we don't change the flush settings, sarama will try to produce messages
+	// as fast as possible to keep latency low.
+	config := sarama.NewConfig()
+	config.Producer.RequiredAcks = sarama.WaitForAll // Wait for all in-sync replicas to ack the message
+	config.Producer.Retry.Max = 10                   // Retry up to 10 times to produce the message
+	config.Producer.Compression = out.GetCompression(codec)
```
Comment (Dieterbe) [inline metric_publish/publish.go]: hmm a little annoying that we're importing fakemetrics/out just to get this utility function. not sure what's an elegant solution. (copypasting the code maybe)

Comment (woodsaj) [inline metric_publish/publish.go]: Looks like raintank/fakemetrics/out should be moved to raintank/metrictank

Comment (woodsaj) [inline metric_publish/publish.go]: may also make sense to rename metrictank/in to metrictank/consume  and metrictank/out to metrictank/publish

Comment (Dieterbe) [inline metric_publish/publish.go]: we decided to not do this just yet https://raintank-corp.slack.com/archives/metrictank/p1468911510000008

**Pull request merged**